discovery-provider/src/tasks/user_replica_set.py

```diff
@@ -80,24 +80,30 @@ def user_replica_set_state_update(
                     # first, get the user object from the db(if exists or create a new one)
                     # then set the lookup object for user_id with the appropriate props
                     if user_id:
-                        existing_user_record = lookup_user_record(
-                            update_task,
-                            session,
-                            entry,
-                            block_number,
-                            block_timestamp,
-                            txhash,
-                        )
+                        if user_id not in user_replica_set_events_lookup:
+                            existing_user_record = lookup_user_record(
```

Comment:
This change will not work

```

user_replica_set_events_lookup[user_id] = {
                                "user": parsed_user_record,
                                "events": [],
                            }
```

We populate the result of parse_user_record into the map *not* the user record itself.

Reply:
but parsed user record is also a user record instance no?